### PR TITLE
Enable SA1000: Keywords should be spaced correctly

### PR DESCRIPTION
**Pull request by xtqqczze**
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1000.md

### CODE REVIEW

**Comment by iSazonov** [comment]
Still draft?

**Comment by xtqqczze** [comment]
@iSazonov Please review and merge.

**Review by iSazonov — APPROVED** [review]

File: test/tools/WebListener/Controllers/ResumeController.cs [hunk 1]
```diff
@@ -113,7 +113,7 @@ private void SetResumeResponseHeaders()
         private bool TryGetRangeHeader(out string rangeHeader)
         {
             var rangeHeaderSv = new StringValues();
```
Comment (iSazonov) [inline test/tools/WebListener/Controllers/ResumeController.cs:115]: @xtqqczze For my education - does an analyzer recognize the "var" pattern to convert to 
```c#
            StringValues rangeHeaderSv = new();
```

Comment (xtqqczze) [reply test/tools/WebListener/Controllers/ResumeController.cs:115]: I am not aware of such an analyzer, but one could use [RCS1012](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1012.md) codefix first.

Comment (iSazonov) [reply test/tools/WebListener/Controllers/ResumeController.cs:115]: So we should apply RCS1012 and then apply a fix to the new pattern?

**Comment by xtqqczze** [comment]
I'm seeing warnings on the following pattern in live analysis, I'm surprised CI did not fail:

`SA1000: The keyword 'new' should be followed by a space.`

https://github.com/PowerShell/PowerShell/blob/db94377b3a69511280d329e64375475f6d7513b8/src/Microsoft.Management.Infrastructure.CimCmdlets/RemoveCimInstanceCommand.cs#L288

This issue was fixed in DotNetAnalyzers/StyleCopAnalyzers#3187, and is in the [v1.2.0-beta.261](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/releases/tag/1.2.0-beta.261) release.

**Comment by iSazonov** [comment]
> I'm surprised CI did not fail:

CI makes clean loading packages. So I guess if you see the warning you need clean your local package cache.

**Comment by xtqqczze** [comment]
> > I'm surprised CI did not fail:
> 
> CI makes clean loading packages. So I guess if you see the warning you need clean your local package cache.

CI is now failing

**Comment by xtqqczze** [comment]
> > @SteveL-MSFT @TravisEz13 those build errors are coming from files I haven't touched in this PR. Not sure what's up there, I'm guessing something got merged that for some reason managed not to trigger the build errors but should have?
> 
> Yeah I'm not able to build this branch locally either.
> 
> Looks like it's coming from [these](https://github.com/PowerShell/PowerShell/pulls?q=is%3Apr+is%3Amerged+IDE0090), but checks for those PRs passed.
> 
> Looking into how best to fix (and ideally prevent from reoccurring).
> 
> /cc @iSazonov

_Originally posted by @rjmholt in https://github.com/PowerShell/PowerShell/issues/9900#issuecomment-736030665_

**Comment by xtqqczze** [comment]
Between tests for this PR passing (12 days ago), we merged [IDE0090](https://github.com/PowerShell/PowerShell/pulls?q=is%3Apr+is%3Amerged+IDE0090) PRs, which conflicted as the version of StyleCopAnalyzers in use at the time did not support target-typed new expressions properly. 

We could reduce the likelihood of similar issues occurring again by running tests again between approval and merge.

**Comment by iSazonov** [comment]
> We could reduce the likelihood of similar issues occurring again by running tests again between approval and merge.

Ah, clear! We catch this again :-)

**Pull request merged**